LLMonFHIR/FHIR Views/FHIRResourcesView.swift

```diff
@@ -124,24 +125,97 @@ struct FHIRResourcesView<ContentView: View, ActionView: View>: View {
     
     
     private func section(for keyPath: KeyPath<FHIRStore, [FHIRResource]>, sectionName: String) -> some View {
-        var resources = fhirStore[keyPath: keyPath]
-        
-        if !searchText.isEmpty {
-            resources = resources.filterByDisplayName(with: searchText)
-        }
-        
+        let resources = filteredResources(for: keyPath)
+
         guard !resources.isEmpty else {
             return AnyView(EmptyView())
         }
-        
+
+        let showAll = Binding(
+            get: { showAllItems[sectionName, default: false] },
+            set: { showAllItems[sectionName] = $0 }
+        )
+
         return AnyView(
-            Section(sectionName) {
-                ForEach(resources.sorted(by: { ($0.date ?? .distantPast) > ($1.date ?? .distantPast) })) { resource in
-                    NavigationLink(value: resource) {
-                        FHIRResourceSummaryView(resource: resource)
-                    }
-                }
+            Section {
+                sectionHeaderButton(sectionName: sectionName, resources: resources, showAll: showAll)
+                resourcesList(resources: resources, showAll: showAll)
+            } header: {
+                EmptyView()
             }
+            .listRowInsets(EdgeInsets(top: 8, leading: 16, bottom: 8, trailing: 16))
         )
     }
+
+    private func filteredResources(for keyPath: KeyPath<FHIRStore, [FHIRResource]>) -> [FHIRResource] {
+        var resources = fhirStore[keyPath: keyPath]
+
+        if !searchText.isEmpty {
+            resources = resources.filterByDisplayName(with: searchText)
+        }
+
+        return resources
+    }
+
+    private func sectionHeaderButton(sectionName: String, resources: [FHIRResource], showAll: Binding<Bool>) -> some View {
+        Button(action: {
+            withAnimation(.spring(response: 0.3, dampingFraction: 0.7)) {
+                showAll.wrappedValue.toggle()
+            }
+        }) {
+            HStack {
+                Text(sectionName)
+                    .font(.headline)
+                    .foregroundColor(.primary)
+
+                countBadge(count: resources.count)
+
+                Spacer()
+
+                if resources.count > 3 {
+                    toggleIndicator(isExpanded: showAll.wrappedValue)
+                }
+            }
+            .contentShape(Rectangle())
+            .padding(.vertical, 6)
+        }
+        .buttonStyle(PlainButtonStyle())
+    }
+
+    private func countBadge(count: Int) -> some View {
```

Comment:
I would get rid of this function and just directly use the `Text` including the view modifiers.